Versatile auth URLs & many UI fixes

venashial opened the pull request:
Spaces out elements in a `.project-card` using `justify-content: space-between;`.

----
_Edit:_ All linked issues:
Fixes modrinth/code#2037
Fixes modrinth/code#2022
Fixes  modrinth/code#2028
Fixes  modrinth/code#2040
Fixes  modrinth/code#2032
Fixes  modrinth/code#2049
Fixes  modrinth/code#2030
Fixes modrinth/code#2026
Fixes modrinth/code#2054
Fixes modrinth/code#2024
Fixes modrinth/code#2020

venashial committed:
Make project cards right-align their last element

Spaces out elements in a `.project-card` using `justify-content: space-between;`.

Fixes modrinth/knossos#170

Geometrically commented:
Could you take a picture of what it looks like now?

wafflecoffee commented:
Doesn't fix the issue
![image](https://i.imgur.com/WIEGHll.png)

venashial commented:
This is what it looks like now:

![Screenshot_2021-05-08 Followed mods - Modrinth](https://user-images.githubusercontent.com/44736536/117556290-0c7ca400-b01c-11eb-9818-4c1df1c0f8a4.png)

venashial commented:
I'm in a local environment and I hard coded some mod IDs to test it.

Logging in kept redirecting me to modrinth.com, and I didn't want to figure out how to get authorization working locally. I probably would have to hard code my token and user ID.

wafflecoffee commented:
`https://api.modrinth.com/api/v1/auth/init?url=http://localhost:3000`

venashial commented:
Thanks! That URL (with my port) works but doesn't persist route changes. I ended up changing this bit in middleware/auth.js to get it to work on all pages:

![image](https://user-images.githubusercontent.com/44736536/117556490-61211e80-b01e-11eb-9a12-c3ec35523bff.png)

venashial commented:
This pull request also fixes the alignment of the settings button on the "My Mods" page:

![image](https://user-images.githubusercontent.com/44736536/117556560-15bb4000-b01f-11eb-8b72-79903fae232e.png)

triphora reviewed (approved):
I am so stupid, I tried running from master branch instead of the patch branch lmao

venashial committed:
Automatically set URL for auth redirect

venashial committed:
Make login button use base url or current origin

Allows the login button to work in dev environment

venashial committed:
Remove Axios base URL trailing slash

venashial committed:
Update authUrl() on dashboard to match default

venashial committed:
Remove 'code' query from URL on page load
Allow non-exact paths to highlight mod & dashboard tabs

Fixes modrinth/knossos#200

venashial committed:
Make page 5 button visible on page 4 (pagination)

Fixes modrinth/knossos#184

venashial committed:
Color links on legal pages

Fixes modrinth/knossos#166

venashial committed:
Set max notifications to 5 and ignore duplicates

Fixes modrinth/knossos#175

venashial committed:
Add space above report button when no user desc

Fixes modrinth/knossos#143

venashial committed:
Better text spacing from edge of mobile screen

Fixes modrinth/knossos#179

venashial committed:
Fix slanted bars in modrinth/knossos#57

venashial committed:
Fix checkbox grid and role label

Fixes modrinth/knossos#191

venashial committed:
Merge branch 'patch-1' of https://github.com/venashial/knossos into patch-1

venashial committed:
Move mod 'settings' button to the far right

Fixes modrinth/knossos#138

wafflecoffee commented on components/layout/MFooter.vue:
```diff
@@ -79,7 +79,7 @@ export default {
 
 footer {
-  padding: 2rem 0 2rem 0;
+  padding: 2rem 1rem;
```
This causes the footer to be very weirdly aligned on the mods pages, dashboard pages, and any other page that uses a thinner footer, on desktop
![image](https://i.imgur.com/aGT4uqN.png)

venashial replied:
My bad, I should have noticed this. I was trying to fix the footer on the index page.

Fixed in aa315876d65c10bc276fe71942e03101481c22a3 using a new prop just for padding.

wafflecoffee commented on components/layout/ModPage.vue:
```diff
@@ -531,7 +533,18 @@ export default {
+    margin-left: auto;
+    @media screen and (max-width: 1024px) {
+      margin-left: 0.5rem;
+      padding-right: 2rem;
```
This is very weird on mobile
![image](https://i.imgur.com/mm6Gk36.png)

venashial replied:
Everything should be looking better with 8411e0d40bdd77abfd7768dceb72c1874f459344.

Old:
![image](https://user-images.githubusercontent.com/44736536/117619483-2a353080-b124-11eb-9046-8f820db8d9e0.png)

Now:
![image](https://user-images.githubusercontent.com/44736536/117619358-040f9080-b124-11eb-96e5-8c79a360c6ac.png)

Mobile:
![image](https://user-images.githubusercontent.com/44736536/117619913-b5aec180-b124-11eb-99a3-afe21321709c.png)

wafflecoffee commented on components/layout/ModPage.vue:
```diff
@@ -490,8 +489,11 @@ export default {
   }
   .info {
     @extend %column;
+    padding: 1rem;
+    display: flex;
+    justify-content: flex-start;
```
![image](https://i.imgur.com/FJYskNd.png)

wafflecoffee replied:
It also looks a bit strange on mobile, with the opposite problem:
![image](https://i.imgur.com/SO4wCuf.png)
The `Mods > Mod Name > Description` should be centred between that rather large gap and that sort of lack of gap

wafflecoffee commented on pages/index.vue:
```diff
@@ -385,6 +385,12 @@ export default {
     text-align: center;
     width: 100%;
+
+    h1,
+    h3,
+    p {
+      padding: 0 5vw;
+    }
```
See image for my review of line 315; this also makes everything look like it's going to the left on larger screens

wafflecoffee replied:
Oddly enough my comment for said line 315 didn't save, lol. 

It still looks strange on 4k screens, albeit not as strange.
![image](https://i.imgur.com/yrVIzWH.png)

venashial replied:
Everything should look better on 4k screens after aed1a2c

wafflecoffee commented on pages/legal/terms.vue:
```diff
@@ -201,4 +201,8 @@ export default {
   @extend %card;
   padding: var(--spacing-card-sm) var(--spacing-card-lg);
 }
+
+a {
+  color: var(--color-link);
+}
```
For consistencies sake, could this also be applied to the about page? It has links underlined but without a colour

venashial replied:
Added in ac33cc730d93aa5b5a75d69bc2288aa5ef40590d

wafflecoffee commented on pages/mod/_id/settings.vue:
```diff
@@ -497,12 +497,30 @@ export default {
 
     .main-info {
       margin-bottom: var(--spacing-card-lg);
+
```
NYF but the adding team members thing overflows on mobile
![image](https://i.imgur.com/LpBihzb.png)

wafflecoffee commented on pages/mods.vue:
```diff
@@ -543,7 +543,7 @@ export default {
 
         this.currentPage = newPageNumber
-        if (pageAmount > 4) {
+        if (pageAmount > 3) {
```
![image](https://i.imgur.com/VOcF59j.png)

The first `-` can be replaced with a 2, not sure how hard that'd be to do

venashial replied:
Should be all fixed

triphora reviewed (changes requested):
Overall, well done! Mostly just minor nitpicks :)

venashial commented:
@wafflecoffee Thanks for the review! I'll fix up the parts needing fixing when I get home.

venashial committed:
Abbreviate minutes to min. when time is too long

Not a perfect solution imo, but works for now

Fixes modrinth/knossos#193

venashial committed:
Fix mobile header margins & add breakpoints

Fixes modrinth/knossos#203

venashial committed:
Clean up nuxt config
Silence babel warning & styleResources

venashial committed:
Upgrade sass-loader to 10.1.1 and remove warning

venashial committed:
Remove added horizontal footer padding

https://github.com/modrinth/knossos/pull/199#discussion_r629011624

wafflecoffee commented on layouts/default.vue:
```diff
@@ -260,7 +292,13 @@ export default {
-        margin-right: 1rem;
+        margin-right: 0.5rem;
+
+        .theme-toggle {
+          @media screen and (max-width: 375px) {
+            display: none;
+          }
```
If I'm reading this right, you're hiding the theme toggle if the screen is smaller than 375px? Doesn't that defeat the whole purpose of the button, to be able to switch modes from mobile?

venashial replied:
It does remove the button, which is the easiest way to make the page fit on mobile phones that small. I was using the logic, "If a phone is that small, then lets hope they have a system wide dark mode toggle like an iPhone 5s." But, that might not be true for other older, smaller phones. Also, maybe a user only wants Modrinth to be in dark mode.

Possible solutions?
- Shrink logo
- Get rid of "Modrinth" wording in logo and just use icon
- Shrink buttons (imo not that great of a solution)

Also, semi-unrelated but, wouldn't it be nice to have the mobile navigation on the bottom of the phone?

wafflecoffee replied:
The second image in #203 is at 360px.

venashial replied:
How did you get that image? 360px looks like this for me (using the changed you suggested):
![image](https://user-images.githubusercontent.com/44736536/117606663-31057880-b10f-11eb-9a0a-7da5058ba7c6.png)

wafflecoffee replied:
That doesn't *look* like 0.5rem to me, but I could be wrong; the way I did it was adding `style="margin-[left or right]:-0.5rem"` to one of the HTML tags lol  
In that case, switching from logo with text to just logo icon makes the most sense under 375px imo

venashial replied:
Everything should be fixed on 8a0a698aa3feca1e8f7ec33bb50aa042de6eb8b3. *This time no dark mode buttons were harmed in the process.*

venashial committed:
Improve mobile header fix

venashial committed:
Fix up minor inconsistencies in mod header

wafflecoffee commented on components/layout/ModPage.vue:
```diff
@@ -465,6 +467,14 @@ export default {
 
       this.userFollows.splice(this.userFollows.indexOf(this.mod.id), 1)
     },
+    formatTime(date) {
+      date = '2021-05-10T03:24:42.106423Z'
+      let defaultMessage = this.$dayjs(date).fromNow()
+      if (defaultMessage.length > 13) {
+        defaultMessage = defaultMessage.replace('minutes', 'min.')
+      }
+      return defaultMessage
+    },
```
No mods were updated recently enough at the time of writing to test this, but, I'm going to assume it works :+1:

Mysterious-Dev commented:
Fabric API is update recently.

wafflecoffee commented:
> Fabric API is update recently. 

Not any within the last hour

Mysterious-Dev commented:
![Screenshot_20210510-152328_Chrome.jpg](https://user-images.githubusercontent.com/40738104/117666022-b8043080-b1a3-11eb-83c8-2e5970432c9d.jpg)

wafflecoffee commented:
Oh. Something's broken then
![image](https://i.imgur.com/DVxfu0Q.png)

Redblueflame commented:
It is maybe caused by the hardcoded date

venashial commented:
Whoops, I forgot to get rid of the hardcoded date I was using to fix modrinth/knossos#193. I'm glad you all caught that.

venashial committed:
Remove hard coded date

venashial committed:
fix merge conflicts

venashial committed:
Merge branch 'modrinth-master' into patch-1

venashial committed:
Cleans up pagination to be more intuitive

venashial committed:
Fixes member invite input on moble

venashial committed:
Fix login button when searching mods

venashial committed:
Improved mobile mod search

Consistently sized pagination buttons

Breakpoint for sort buttons on smaller screens

venashial committed:
Consistent link style on text-only pages

venashial committed:
Better 4k support

venashial committed:
Slightly better mobile project-card support

Shuffles categories under mod icon when there is room

venashial committed:
Animate homepage typewriter effect backwards

venashial committed:
Tiny commit to align mod icons in mod headers

venashial committed:
Make processing status include 'Under Review'

This can be later updated once the backend has a separate status

venashial committed:
Create vercel.json

Redblueflame reviewed (approved):
That was quite a big PR, in the future please make multiple smaller ones.
But all in all, looks good to me, thanks for your contribution 👍

venashial commented:
You are totally right, this PR just grew and grew. I wasn't sure if the lack of merge conflicts would outweigh the mess of it all.

venashial commented:
I don't think this should be merged yet. There are issues with the sign-in button being given a broken URL (a localhost one) when rendered server-side with Heroku. The server doesn't have a way of knowing the page domain. I think this should be fixed by having some variable for the page domain (maybe in the vuex store) that is either localhost:port when in development or a set value in production.

venashial committed:
Update domain auto detection

venashial committed:
Merge branch 'master' into patch-1

venashial committed:
Add vercel.json

venashial committed:
Test vercel NODE_ENV

venashial commented:
Still not working, something to do with NODE_ENV I think.

venashial commented:
Seems to work locally and on Heroku & Vercel now. I'll just remove the console.log() now.

venashial committed:
Remove console.log for debugging hosting services

venashial commented:
The Vercel page: https://knossos-preview-7ercbw7ft-venashial.vercel.app/

venashial commented:
Small note: The URL is automatically picked using Heroku env variables that are only injected in pull request preview deployments. That's why there is still a fallback to `'https://modrinth.com'`. If [dyno metadata](https://devcenter.heroku.com/articles/dyno-metadata) was enabled, it wouldn't be needed.

triphora reviewed (approved):
Awesome, couldn't find any issues in my testing :)

falseresync commented on assets/images/utils/settings.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="12" cy="12" r="3"></circle><path d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1. ...
```
Use this to optimize https://jakearchibald.github.io/svgomg/

venashial replied:
Done in a1c9a5c

falseresync commented on components/ui/ProjectCard.vue:
```diff
@@ -40,7 +41,7 @@
                 >
                 <span v-if="status === 'processing'" class="badge yellow">
-                  Processing
+                  Processing/Under Review
```
An `or` is more visually appealing. Additionally, try not to use Pascal Case. It is out of style, plus just a tiny bit harmful to accessibility if used too much

venashial replied:
I can use an `or`, it just makes the already large badge larger. I like having `Review` capitalized, but I'll change it.

venashial replied:
Neither `or` nor Sentence case look more visually appealing to me after trying them.

Which of the options below do you like the most? Or did you have something else in mind?

\#&#x2060;1
![image](https://user-images.githubusercontent.com/44736536/118195615-63112600-b400-11eb-83d8-80044fdd5bcd.png)
\#&#x2060;2
![image](https://user-images.githubusercontent.com/44736536/118196060-34e01600-b401-11eb-9684-515112558c3b.png)
\#&#x2060;3
![image](https://user-images.githubusercontent.com/44736536/118195674-84721200-b400-11eb-8699-b97f27486716.png)
\#&#x2060;4
![image](https://user-images.githubusercontent.com/44736536/118195727-9a7fd280-b400-11eb-964f-edcc580e411e.png)
\#&#x2060;5
![image](https://user-images.githubusercontent.com/44736536/118195785-b97e6480-b400-11eb-8bb9-2c22b5d4e711.png)

Personally, I like #&#x2060;6. #&#x2060;3 is okay too.

wafflecoffee replied:
Personally, I'd just prefer `Under review`, no need to even have `Processing` there imo, but false may have something else in mind

Minenash replied:
If I remember correctly, the Processing was for when Modrinth checked files automatically, but seeing that doesn't exist yet, won't be a thing for a while, and should probably have it's own status, I agree it should just be `Under review`

venashial replied:
I agree, `Under review` makes the most sense currently. If/when the backend starts processing files, this will be easy to change.

falseresync commented on layouts/default.vue:
```diff
@@ -230,19 +259,36 @@ export default {
+        .small-logo {
+          display: none;
+        }
+        @media screen and (max-width: 350px) {
```
No max-width, be mobile-first: use min-width

venashial replied:
Fixed with fa1e60c

falseresync commented on pages/dashboard.vue:
```diff
@@ -78,7 +78,9 @@ export default {
     authUrl() {
-      return `https://api.modrinth.com/api/v1/auth/init?url=https://modrinth.com${this.$route.fullPath}`
+      return `https://api.modrinth.com/api/v1/auth/init?url=${
+        process.env.domain || window.location.origin
```
Are you sure this is right logic here? Perhaps null coalescing operator is better (?? operator). || operator will default to the right hand side on any falsey value, including `''` and `0`, while ?? operator will only do so on `null` or `undefined` value

venashial replied:
It doesn't really matter, because `process.env.domain` would never be `''` or `0`. Though, maybe it isn't even needed. [This](https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-env/) makes it sound like environment variables are available client side.

venashial replied:
Turns out `window.location.origin` isn't needed. I removed it in b44fdc0.

venashial committed:
Make mobile first + fix shrinked text circle size

venashial committed:
Optimize SVG

venashial committed:
Change media queries to be more mobile first

venashial committed:
Remove `|| window.location.origin`

venashial commented:
Updated Vercel deployment: https://knossos-preview-ng7yipbx0-venashial.vercel.app/

Mysterious-Dev commented:
It's a good work.

venashial committed:
Merge branch 'modrinth:master' into patch-1

venashial committed:
re-deploy vercel

venashial commented:
Everything is finally merged with upstream correctly.

Mysterious-Dev commented:
@falseresync review ?

venashial committed:
Merge branch 'modrinth:master' into patch-1

venashial committed:
Change "Processing" message to "Under review"

WickedTree commented:
Dunno if this should be put in its own PR, but alignment for the version list is kind of funky: 
![image](https://user-images.githubusercontent.com/50127946/119690405-82478480-bdfe-11eb-88b8-5be89bddc23d.png)
especially the mod loader and status columns.
Maybe just center everything?
there also isn't enough space between "date published" and the edge

venashial commented:
> Dunno if this should be put in its own PR, but alignment for the version list is kind of funky:

@WickedTree 

Probably could be its own PR, related to #219.
